Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -343,6 +357,24 @@ static void __GetGlyphRunDetails(_CTTypesetter* ts, CFRange range, CGRect frameS
 }
 
 /**
+ * Helper method to create _CTLine object given a CFAttributedStringRef
+ *
+ * @parameter string CFAttributedStringRef containing text and styling
+ *
+ * @return Unmutable array of _CTLine objects created with the requested parameters.
+ */
+static _CTLine* _DWriteGetLine(CFAttributedStringRef string) {
+    _CTTypesetter* typesetter = static_cast<_CTTypesetter*>(CTTypesetterCreateWithAttributedString(string));
+    CFRange range = CFRangeMake(0, CFAttributedStringGetLength(string));
+    NSArray<_CTLine*>* lines = _DWriteGetLines(typesetter, range, CGRectMake(0, 0, FLT_MAX, FLT_MAX));
+    if (lines.count > 0) {
+        return [lines[0] retain];
+    }
+
+    return [_CTLine new];
```

Comment:
The reference platform will create an empty line if given "", and expects a non-null argument for CTLineCreateWithAttributedString

---
In reply to: [79685824](https://github.com/Microsoft/WinObjC/pull/989#discussion_r79685824) [](ancestors = 79685824)